src/seraphis_main/enote_record_types.h

```diff
@@ -129,32 +129,28 @@ struct LegacyEnoteRecord final
 
 ////
 // SpBasicEnoteRecordV1
-// - a seraphis enote that has passed the view-tag check using a jamtis find-received key
+// - a seraphis enote that resides in a transaction with an enote that passes the primary view tag check
 ///
-struct SpBasicEnoteRecordV1 final
+struct SpBasicEnoteRecordV1
 {
     /// original enote
     SpEnoteVariant enote;
     /// the enote's ephemeral pubkey
     crypto::x25519_pubkey enote_ephemeral_pubkey;
+    // the transaction primary view tag size
+    std::uint8_t num_primary_view_tag_bits;
     /// context of the tx input(s) associated with this enote
     rct::key input_context;
-    /// t'_addr: nominal address tag (only useful for jamtis non-selfsend enote types)
-    jamtis::address_tag_t nominal_address_tag;
+    /// did this enote not pass the check?
+    bool is_auxiliary;
 };
 
 ////
 // SpIntermediateEnoteRecordV1 (jamtis non-selfsend enote type only)
-// - a seraphis enote with info extracted using a jamtis find-received key, generate-address secret, and unlock-amounts key
+// - a seraphis enote with info extracted using a jamtis view-received key and generate-address secret key
 ///
-struct SpIntermediateEnoteRecordV1 final
+struct SpIntermediateEnoteRecordV1: public SpBasicEnoteRecordV1
```

Comment:
Would it be better to nest the structs such that intermediate enote record has a member field (e.g. `basic`), and the full enote record has an intermediate record as a member field (e.g `intermediate`)?

Reply:
IIRC, writing the structs like this made the enote record utils code more compact, which makes sense since the information in each is a superset of the previous. I will revert tho

Reply:
Okay done